src/printer/index.js

```diff
@@ -305,7 +305,7 @@ function genericPrint(path, options, print) {
       const parent = path.getParentNode();
```

Comment:
Instead of reaching to the parent here, just drop `visit_field_Num_n` from `astexport`, then just print `path.source` in the handler for `Num`. As noted above, this is the right thing to do for complex values too.

In general it's probably best not to try to normalize numbers in Python. People do lots of numerical work, and if I'm writing out something like `1.00`, it's probably for a good reason – like maybe it's `np.array([0.99, 1.00, 1.01])` or something.

Reply:
Can't quite do that as `complex` is not JSON serializable. Having the source type information may be useful to, so I'll just coerce them to `str` on the Python side (assuming that is safe?).